sr3 doesn't declare exchanges (maybe users/sources too?) from admin.conf if they're not used in a config

config:

```
# created by: sr3 convert winnow/rs_testing
vip 1.2.3.4

logrotate 3
#loglevel debug
nodupe_ttl 900s
nodupe_basis name

broker amqps://SOME-SOURCE@ddsr-dev/
exchange xs_SOME-SOURCE_WINNOW

queueName q_${BROKER_USER}.${PROGRAM}.${CONFIG}.ddsr-dev


post_broker amqps://SOME-SOURCE@ddsr-dev/
post_exchange xs_SOME-SOURCE

instances 1
identity arbitrary
```

```
2024-01-25 21:33:26,088 [ERROR] sarracenia.moth.amqp getSetup AMQP getSetup failed to ddsr-dev with Queue.bind: (404) NOT_FOUND - no exchange 'xs_SOME-SOURCE_WINNOW' in vhost '/'
```

sr3d show does show that exchangeDeclare is True.

I'm not sure if it's supposed to declare the source exchange or not? Maybe it's only supposed to declare the post_exchange?

This might apply to other components that have a (source) exchange and post_exchange like a sarra. I haven't had time to test.

**UPDATE**: a good way to address this is to have sr3 honour exchange declarations in the admin.conf file.  This issue is now about that.

--- a/sarracenia/sr.py
+++ b/sarracenia/sr.py
@@ -301,15 +301,6 @@ class sr_GlobalState:
 
         self.default_cfg = sarracenia.config.default_config()
 
-        #self.default_cfg = sarracenia.config.Config()
-        #if os.path.exists( "default.conf" ):
-        #    self.default_cfg.parse_file("default.conf")
-        #if os.path.exists( "admin.conf" ):
-        #    self.default_cfg.parse_file("admin.conf")
-
-        #self.admin_cfg = copy.deepcopy( self.default_cfg )
-        #if os.path.exists( "admin.conf" ):
-        #    self.admin_cfg.parse_file("admin.conf")
         os.chdir(self.user_config_dir)
 
         for c in self.components:
@@ -1361,6 +1352,20 @@ class sr_GlobalState:
                                     self.default_cfg.declared_users[u_url.username],
                                     u_url.username, u_url.password, self.options.dry_run )
 
+        # declare admin exchanges.
+        if hasattr(self,'default_cfg'):
+            logger.info( f"Declaring exchnges for admin.conf using {self.default_cfg.admin} ")
+            if hasattr(self.default_cfg, 'declared_exchanges'):
+                xdc = sarracenia.moth.Moth.pubFactory(
+                    {
+                        'broker': self.default_cfg.admin,
+                        'dry_run': self.options.dry_run,
+                        'exchange': self.default_cfg.declared_exchanges,
+                        'message_strategy': { 'stubborn':True }
+                    })
+                xdc.putSetup()
+                xdc.close()
+               
         # declare exchanges first.
         for f in self.filtered_configurations:
             if self.please_stop:
@@ -1426,7 +1431,6 @@ class sr_GlobalState:
                 flow.runCallbacksTime('on_declare')
                 del flow
                 flow=None
-        
 
     def disable(self):
         if len(self.filtered_configurations) == 0:

